qf_lib/data_providers/sp_global/sp_data_provider.py

```diff
@@ -0,0 +1,353 @@
+
+        for field in [f for f in fields if f not in price_fields]:
+            try:
+                dfs.append(field_to_fetching_method[field])
+            except KeyError:
+                raise ValueError(f"Field {field} is not supported by the SPDataProvider.") from None
+
+        grouped = pd.concat(dfs, axis=1).groupby(level=0)
+        tickers_dict = {tid_to_tickers[tid]: group.droplevel(0) for tid, group in grouped}
+        data_array = tickers_dict_to_data_array(tickers_dict, tickers, fields)
+        return normalize_data_array(data_array, tickers, fields, got_single_date, got_single_ticker, got_single_field)
+
+    def get_currency(self, tickers: Union[SPTicker, Sequence[SPTicker]]) -> Union[str, QFSeries]:
+        """
+        Function returning currency for the given selection of SPTickers.
+
+        Parameters
+        -----------
+        tickers: SPTicker, Sequence[SPTicker]
+            tickers of securities for which currencies should be retrieved
+
+        Returns
+        --------
+        str, QFSeries
+            currency for the given tickers
+        """
+
+        tickers, got_single_ticker = convert_to_list(tickers, SPTicker)
+        currencies = QFSeries()
+        sp_id_to_ticker = {t.tradingitem_id: t for t in tickers}
+
+        with self._db_connection_provider.Session.begin() as session:
+            for ticker_group in grouper(1000, sp_id_to_ticker.keys()):
+                sql_statement = session.query(self.ciqtradingitem.tradingitemid, self.ciqcurrency.isocode) \
+                    .join(self.ciqcurrency, self.ciqtradingitem.currencyid == self.ciqcurrency.currencyid) \
+                    .filter(self.ciqtradingitem.tradingitemid.in_(ticker_group)).statement
+
+                index_col = self.ciqtradingitem.tradingitemid.key
+                currency = QFSeries(pd.read_sql(sql_statement, con=session.bind, index_col=index_col).iloc[:, 0])
+                currency = currency.rename(index=sp_id_to_ticker)
+                currencies = pd.concat([currencies, currency])
+
+        currencies = currencies.reindex(tickers).replace({np.nan: None})
+        return currencies[0] if got_single_ticker else currencies
+
+    def supported_ticker_types(self) -> Set[Type[Ticker]]:
+        return {SPTicker}
+
+    def _fetch_in_chunks(fetch_func):
+        """
+        Utility decorator, which batches the tickers before entering the querying function, maps all the fields
+        into SPField objects and returns a list of all dataframes. It is assumed that the decorated function
+        returns a dataframe.
+        """
+        @wraps(fetch_func)
+        def wrapper(self, tickers, *args, **kwargs):
+            # Expecting identifiers to be the first argument
+            results = []
+            for ticker_group in grouper(1000, tickers):
+                data_frame = fetch_func(self, ticker_group, *args, **kwargs)
+                data_frame = data_frame.rename(columns={f.value: f for f in SPField})
```

Comment:
Here you will fetch all values in SPField, but the dataframe will only have the fields specified in the `fields` parameter of `get_history()`. 
+ We will probably add more to the enums and the order might change etc.